### ISSUE TITLE
Typescript support

### ISSUE THREAD

**Issue by JonasKruckenberg** [hyperswarm/hyperswarm#55]
I really like this module, I love that it is way simpler to use and understand than many other solutions out there. But I wish it would work better with typescript and all the other tooling out there. 
That's why I started working on 'porting' Hyperswarm to typescript on my fork, it's not yet complete and I'm probably also not the best typescript dev in existence, but it's a starting point that I use in my project.
Since I think other users of the module might benefit from stuff that I've been doing i.e. typings, docs, etc. I thought I ask you what the wisest thing to do would be.
Are you open to accepting a .d.ts file PR? Or should my fork be kept separate?
I've generated a bit of WIP docs from the code as well, maybe that can be put to use somehow.
I'm here to help make this module more accessible if that's possible!
Cheers!

**Comment by davidmarkclements** [comment]
hey @JonasKruckenberg would you be willing to maintain a d.ts file long term? 

I personally think d.ts files are great with just normal JS because of the intellisense benefits and ease integration for those who do use typescript. The only issue is that TypeScript moves forward as a language quite quickly so the d.ts files can quickly become outdated, not because of breaking library changes but because of breaking TypeScript changes. This is what I found with Pino anyway, eventually we removed the d.ts file as it became too contentious. 

But if you're okay to commit to maintaining d.ts files each of the hyperswarm modules then I'm +1 on accepting a PR for a d.ts file on each repo

**Comment by JonasKruckenberg** [comment]
Yeah I think the majority of the work isn’t to maintain the typings but to create them in the first place, so yeah I‘m willing to maintain the typings long term. I’d could use some help creating explanations for the functions though, but I can count on your help there right?

What modules are you considering part of Hyperswarm that are worth getting typings? Hyperswarm, Discovery, Network, Dht and replicator are ones I was thinking of 🤔

**Comment by robbiemu** [comment]
this still is a wanted feature!

```
robertotomas@MacBook-Pro-2 ttt-p2p % yarn add --dev @types/hyperswarm
yarn add v1.22.19
[1/4] 🔍  Resolving packages...
error An unexpected error occurred: "https://registry.yarnpkg.com/@types%2fhyperswarm: Not found".
info If you think this is a bug, please open a bug report with the information provided in "/Users/robertotomas/Public/ttt-p2p/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
```

**Comment by deanpress** [comment]
+1 for typescript support!

**Comment by eL1x00r** [comment]
guys big thing, wen ?

**Comment by executionreverted** [comment]
+1. Also typescript react/any other framework boilerplates would be nice

**Comment by princecodes247** [comment]
+1 for typescript support

**Comment by smoyer64** [comment]
I'm not sure this library should be polluted by framework code - how about a separate package or packages?
